test/tools/fuzzTesting/createRandomTest.cpp

```diff
@@ -42,71 +42,29 @@ int checkRandomTest(std::function<void(json_spirit::mValue&, bool)> _doTests, js
 namespace dev { namespace test {
 int createRandomTest(std::vector<char*> const& _parameters)
 {
-	std::string testSuite;
-	std::string testFillString;
-	json_spirit::mValue testmValue;
-	bool checktest = false;
-	bool filldebug = false;
 	bool debug = false;
-	bool filltest = false;
-
+	TestOutputHelper::initTest(1);
 	dev::test::Options& options = const_cast<dev::test::Options&>(dev::test::Options::get());
 
-	testSuite = options.rCurrentTestSuite;
-	if (testSuite != "BlockChainTests" && testSuite != "TransactionTests" && testSuite != "StateTests"
-				&& testSuite != "VMTests")
-		testSuite = "";
+	std::string const& testSuite = options.rCurrentTestSuite;
+	if (testSuite != "BlockchainTests" && testSuite != "TransactionTests" && testSuite != "StateTestsGeneral"
+		&& testSuite != "VMTests")
+	{
+		std::cerr << "Error! Test suite '" + testSuite +"' not supported! (Usage -t TestSuite)" << std::endl;
+		return 1;
+	}
 
+	//parse options for createRandomTest special test suite
 	for (size_t i = 0; i < _parameters.size(); ++i)
 	{
 		auto arg = std::string{_parameters.at(i)};
 
-		if (arg == "--fulloutput")
-			options.fulloutput = true;
-		else if (arg == "--debug")
+		if (arg == "--debug")
 			debug = true;
-		else
-		if (arg == "--filldebug")
-			filldebug = true;
 	}
 
-	if (testSuite == "")
-	{
-		std::cerr << "Error! Test suite not supported! (Usage -t TestSuite)" << std::endl;
-		return 1;
-	}
-	else
-	{
-		if (checktest)
-			std::cout << "Testing: " << testSuite.substr(0, testSuite.length() - 1) << "... ";
-
-		if (testSuite == "BlockChainTests")
-		{
-			if (checktest)
-				return checkRandomTest(dev::test::doBlockchainTests, testmValue, debug);
-			else
-				fillRandomTest(dev::test::doBlockchainTests, (filltest) ? testFillString : c_testExampleBlockchainTest, filldebug);
-		}
-		else
-		if (testSuite == "TransactionTests")
-		{
-			if (checktest)
-				return checkRandomTest(dev::test::doTransactionTests, testmValue, debug);
-			else
-				fillRandomTest(dev::test::doTransactionTests, (filltest) ? testFillString : c_testExampleTransactionTest, filldebug);
-		}
-		else
-		if (testSuite == "VMTests")
-		{
-			if (checktest)
-			{
-				dev::eth::VMFactory::setKind(dev::eth::VMKind::JIT);
-				return checkRandomTest(dev::test::doVMTests, testmValue, debug);
-			}
-			else
-				fillRandomTest(dev::test::doVMTests, (filltest) ? testFillString : c_testExampleVMTest, filldebug);
-		}
-	}
+	if (testSuite == "StateTestsGeneral")
+		fillRandomTest(dev::test::doStateTests, c_testExampleStateTest, debug);
```

Comment:
Why doesn't VMTest appear here?  Maybe we should say "Test suite VMTest not supported" above?